bindings/cs/Scripts/build.cake

```diff
@@ -0,0 +1,218 @@
+#addin nuget:?package=Cake.FileHelpers&version=3.3.0
+#addin nuget:?package=Cake.Git&version=0.22.0
+#addin nuget:?package=Cake.Incubator&version=5.1.0
+#addin nuget:?package=Cake.Compression&version=0.2.4
+#tool dotnet:?package=GitVersion.Tool&version=5.3.7
+#tool nuget:?package=xunit.runner.console
+
+using System;
+using System.Linq;
+
+var target = Argument("target", "CreatePackage");
+var configuration = Argument("configuration", "Release");
+
+var gitVersion = GitVersion();
+
+var buildDirectory = MakeAbsolute(Directory(GetSettingValue("buildDirectory", "c:\\build")));
+var libmongocryptAllDirectory=buildDirectory.Combine("libmongocrypt-all");
+var downloadedMongocryptDirectory=buildDirectory.Combine("downloadedMongocryptDirectory");
+var localReleaseVersion = "local-0.0.0";
+var releaseVersion = GetSettingValue("releaseVersion", localReleaseVersion);
+var fork = GetSettingValue("fork", "https://github.com/mongodb/libmongocrypt.git");
+var branch = GetSettingValue("branch", "master"); 
+var libmongocryptAllUrl = GetSettingValue("url", "https://mciuploads.s3.amazonaws.com/libmongocrypt/all/master/508e21f4abff9f5519e0357a63a4ad30d2c24692/libmongocrypt-all.tar.gz");
+var csharpBindingsGitTagName = $"csharp-v{releaseVersion}";
+var csharpBindingsDirectory = buildDirectory.Combine(csharpBindingsGitTagName);
+var libmongocryptRelWithDebInfoDirectory = csharpBindingsDirectory.Combine("cmake-build").Combine("RelWithDebInfo");
+var libmongocryptCsDirectory = csharpBindingsDirectory.Combine("bindings").Combine("cs");
+var libmongocryptSolutionDirectory = libmongocryptCsDirectory.Combine("MongoDB.Libmongocrypt");
+var libmongocryptSolutionFile = libmongocryptSolutionDirectory.CombineWithFilePath("MongoDB.Libmongocrypt.csproj");
+var libmongocryptTestsSolutionDirectory = libmongocryptCsDirectory.Combine("MongoDB.Libmongocrypt.Test");
+var artifactsDirectory = buildDirectory.Combine("artifacts");
+
+Task("Prepare")
+    .Does(() =>
+    {
+        if (DirectoryExists(buildDirectory))
+        {
+            DeleteDirectory(
+                buildDirectory, 
+                new DeleteDirectorySettings {
+                    Recursive = true,
+                    Force = true
+                });
+        }
+        CreateDirectory(buildDirectory);
+
+        Information($"Release version: {releaseVersion}");
+        Information($"Fork: {fork}");
+        Information($"Branch: {branch}");
+        Information($"Native libraries url: {libmongocryptAllUrl}");
+
+        Information("Downloading native libs..");
+        EnsureDirectoryExists(libmongocryptAllDirectory);
+        var nativeLibrariesArchive = libmongocryptAllDirectory.CombineWithFilePath("libmongocrypt-all.tar");
+        DownloadFile(libmongocryptAllUrl, nativeLibrariesArchive);
+        
+        Information("Unzipping..");
+        UncompressToTheCurrentDirectory(nativeLibrariesArchive);
+
+        Information("Cloning the libmongocrypt repo..");
+        GitClone(
```

Comment:
should we consider some debugging case when we didn't clone the whole repo but copy the current working directory?

Reply:
in this case we can use `gitVersion.LegacySemVer` instead of `var localReleaseVersion = "local-0.0.0";`

Reply:
because in local development (and as the result in a local solution), we don't have control (via git branches) over which native libraries are copied into the `cmake-build` folder, so if we're working on some ticket, then if we checkout on the master, the native libraries in `cmake-build` folder won't be changed, we can always clean them in each release, but then it can broke the local ticket

Reply:
Fair enough.